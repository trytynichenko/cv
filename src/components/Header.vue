<template>
  <header class="header block">
    <navigation :items="nav"/>
  </header>
</template>

<script>
import Navigation from '@/components/Navigation.vue';

export default {
  name: 'Header',
  components: {
    Navigation,
  },
  data: () => ({
    nav: [
      { name: 'Home', path: '/' },
      { name: 'About', path: '/about' },
      { name: 'Contact', path: '/contact' },
    ],
  }),
};
</script>

<style scoped lang="scss">
  @import '../assets/styles/components/header';
</style>
