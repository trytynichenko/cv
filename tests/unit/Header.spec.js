import { shallow } from '@vue/test-utils';
import Header from '@/components/Header.vue';

describe('Header.vue', () => {
  const props = {
    items: [
      { name: 'item1', path: '/item1' },
      { name: 'item2', path: '/item2' },
      { name: 'item3', path: '/item3' },
    ],
  };

  const wrapper = shallow(Header, {
    propsData: props,
  });

  it('matches snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
