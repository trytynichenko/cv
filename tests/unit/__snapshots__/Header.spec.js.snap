// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header.vue matches snapshot 1`] = `
<header class="header block">
    <!---->
</header>
`;
